### PR TITLE
Added Digishield support and unit tests

### PR DESCRIPTION
**Pull request by rnicoll**
Having problems with the unit tests. Creating PR now for visibility, but please don't merge until either the tests work, or it's clear that they're broken by something still pending.

### CODE REVIEW

File: src/test/dogecoin_tests.cpp [hunk 1]
```diff
@@ -96,10 +96,99 @@ BOOST_AUTO_TEST_CASE(subsidy_limit_test)
...
+    SelectParams(CBaseChainParams::MAIN);
+    const Consensus::Params& params = Params().GetConsensus();
+
+    CBlockIndex pindexLast;
+    int64_t nLastRetargetTime = 1386474927; // Block # 1
+    
+    pindexLast.nHeight = 239;
+    pindexLast.nTime = 1386475638; // Block #239
+    pindexLast.nBits = 0x1e0ffff0;
+    BOOST_CHECK_EQUAL(CalculateNextWorkRequired(&pindexLast, nLastRetargetTime, params), 0x1e00ffff);
```
Comment (langerhans) [inline src/test/dogecoin_tests.cpp]: If you change all these to `CalculateDogecoinNextWorkRequired` all tests will pass ;)

**Comment by langerhans** [comment]
ACK pending test fixes

**Comment by rnicoll** [comment]
Thanks for spotting that, have updated the tests now.

**Comment by patricklodder** [comment]
works for me too, tests pass as well now. merging this.

**Pull request merged**